src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -383,21 +383,21 @@ def _await_fluent_launch(
 
 
 def _get_server_info(
-    server_info_file_name: str,
+    server_info_filepath: Optional[str] = None,
     ip: Optional[str] = None,
     port: Optional[int] = None,
     password: Optional[str] = None,
 ):
     """Get server connection information of an already running session."""
-    if ip and port:
+    if ip and port and server_info_filepath:
         logger.warning(
             "Could not parse server-info file because ip and port were provided explicitly."
         )
     elif server_info_file_name:
         ip, port, password = _parse_server_info_file(server_info_file_name)
     elif os.getenv("PYFLUENT_FLUENT_IP") and os.getenv("PYFLUENT_FLUENT_PORT"):
         ip = port = None
-    else:
+    elif not (ip and port) and not server_info_filepath:
```

Comment:
It is a bit difficult to understand the whole if..else..if nesting. The code would be more readable if it is linearized:
```
if wrong combination 1:
    show warning or error
if wrong combination 2:
    show warning or error
parse server info file if file is specified
ip = from file or from env var or default
port = from file or from env var or default
password = from file
```

Reply:
💯